Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM haskell:8.6 as build
+WORKDIR /build
+RUN cabal new-update
+
+# Build our upstream dependencies after copying in only enough to tell cabal
+# what they are.  This will make these layers cache better even as we change the
+# code of semantic itself.
+COPY semantic.cabal /build/
```

Comment:
Since the `WORKDIR` was set on line 2, this and the other copies can be just:

`COPY semantic.cabal .`

Reply:
Done